src/main/java/org/folio/rest/utils/MigrationUtils.java

```diff
@@ -0,0 +1,151 @@
+    if (moduleTo.compareTo(currentModuleVersion) > 0) {
+      log.info("migrate:: moduleTo={} was greater than currentModuleVersion={} version", moduleTo, currentModuleVersion);
+      return supplier.get();
+    }
+
+    log.info("migrate:: moduleTo={} was less or equals to currentModuleVersion={}", moduleTo, currentModuleVersion);
+    return Future.succeededFuture();
+  }
+
+  public static SemVer moduleVersionToSemVer(String version) {
+    try {
+      return new SemVer(version);
+    } catch (IllegalArgumentException ex) {
+      return new ModuleId(version).getSemVer();
+    }
+  }
+
+  public static Future<Void> updateInvoiceWithFiscalYear(Map<String, String> headers, Context vertxContext) {
+    DBClient dbClient = new DBClient(vertxContext, headers);
+    RequestContext requestContext = new RequestContext(vertxContext, headers);
+    PostgresClient pgClient = dbClient.getPgClient();
+    return pgClient.withTrans(conn -> getInvoicesWithoutFiscalYearFromDb(conn)
+      .compose(invoices -> getTransactionsByInvoiceIdsAndPopulate(invoices, requestContext))
+      .compose(invoices -> updateInvoicesWithFiscalYearId(invoices, conn)))
+      .onSuccess(v -> log.info("updateInvoiceWithFiscalYear:: Updating invoices with fiscalYearId fully completed"))
+      .onFailure(t -> log.info("Error to update invoices with fiscalYear", t));
+  }
+
+  private static Future<List<Invoice>> getInvoicesWithoutFiscalYearFromDb(Conn conn) {
+    Criteria criteria = new Criteria();
+    criteria.addField("'fiscalYearId'");
+    criteria.setOperation("IS NULL");
+    Criterion criterion = new CriterionBuilder().build();
+    criterion.addCriterion(criteria);
+
+    return conn.get(INVOICE_TABLE, Invoice.class, criterion, false)
+      .map(Results::getResults)
+      .onFailure(t -> log.error("Failed to fetch invoice with query={}", criterion.toString(), t));
+  }
+
+  private static Future<List<Invoice>> getTransactionsByInvoiceIdsAndPopulate(List<Invoice> invoices, RequestContext requestContext) {
+    if (invoices.isEmpty()) {
+      log.info("getTransactionsByInvoiceIds:: No invoices to process");
+      return Future.succeededFuture();
+    }
+
+    // Extract the invoice IDs from the list of invoices
+    List<String> invoiceIds = invoices.stream()
+      .map(Invoice::getId)
+      .toList();
+
+    return retrieveTransactionsByInvoiceIds(invoiceIds, requestContext)
+      .compose(response -> populateInvoicesWithFiscalYears(invoices, response));
+  }
+
+  private static Future<Void> updateInvoicesWithFiscalYearId(List<Invoice> invoices, Conn conn) {
+    if (Objects.isNull(invoices)) {
+      return Future.succeededFuture();
+    }
+    return conn.updateBatch(INVOICE_TABLE, invoices)
+      .onComplete(results -> log.info("updateInvoicesWithFiscalYearId:: Successfully '{}' invoice(s) updated", invoices.size()))
+      .onFailure(e -> log.error("Error to update '{}' invoice(s) in INVOICE_TABLE '{}'", invoices.size(), INVOICE_TABLE))
```

Comment:
need to add exception stack trace to log

Reply:
added, thanks